pkg/cmd/roachtest/test_impl.go

```diff
@@ -333,7 +333,7 @@ func (t *testImpl) addFailure(format string, args ...interface{}) {
 	if format == "" {
 		format = strings.Repeat(" %v", len(args))[1:]
 	}
-	reportFailure := newFailure(errors.NewWithDepthf(1, format, args...), collectErrors(args))
+	reportFailure := newFailure(errors.NewWithDepthf(2, format, args...), collectErrors(args))
```

Comment:
It's not immediately obvious why `2` is better than `1`. Presumably, this is so that `runtime.Callers` will skip this frame (and the ones below it)? It might be worth explaining this in a comment; personally, I find the depth arithmetic error-prone.

Reply:
The depth here tells us how many of the most recent frames to ignore. That should give us the, in the logs, where `t.Fatalf` was called from (which can be from roachtest internal, or an actual roachtest).

Previously, we were creating this error from `t.Fatalf`, and the depth was 1. During more recent refactoring, the error creation was moved to this shared `addFailure`, which is 1 more call away, but the depth was not incremented to account for that.